<launch>

<arg name="map_size_x" default="20.0"/>
<arg name="map_size_y" default="20.0"/>
<arg name="map_size_z" default="2.0" />
<arg name="poly_order" default=" 5"/>
      
  <node pkg = "trr_local_replanner" type = "local_replanner" output = "screen" name = "local_replanner">
      <rosparam file="$(find trr_local_replanner)/config/generator_replan.yaml" command="load" />
      <remap from = "~replan_mission" to = "/time_optimal_traj_server/check_pts"/>
     
      <remap from = "/sdf_map/depth"  to = "/camera/depth/image_rect_raw"/>
      <remap from = "/sdf_map/pose"   to = "/vins_estimator/camera_pose"/> 
      <remap from = "/sdf_map/odom"   to = "/drone_1/visual_slam/odom"/> 
      <remap from = "/sdf_map/cloud"  to = "/pcl_render_node/local_pointcloud"/> 

      <param name="local_planner/vis/vis_traj_width"    value="0.2" /> 
      <param name="local_planner/vis/esdf_grid_freq"    value="0.1" /> 
      <param name="local_planner/vis/frame_id"          value="/map" /> 
      
      <param name="local_planner/time_emergency"        value="0.5" /> 
      <param name="local_planner/max_vel"               value="2.5" /> 
      <param name="local_planner/max_acc"               value="2.5" /> 

      <!-- <param name="sdf_map/input_data_type"    value="$(arg sense_data_type)"/> -->

      <param name="sdf_map/resolution"      value="0.1" /> 
      <param name="sdf_map/global/x_size"   value="$(arg map_size_x)" /> 
      <param name="sdf_map/global/y_size"   value="$(arg map_size_y)" /> 
      <param name="sdf_map/global/z_size"   value="$(arg map_size_z)" /> 
      <param name="sdf_map/local_radius_x"  value="4.5" /> 
      <param name="sdf_map/local_radius_y"  value="4.5" /> 
      <param name="sdf_map/local_radius_z"  value="5.5" /> 
      <param name="sdf_map/inflate_val"     value="0.2" /> 
      <param name="sdf_map/esdf_inflate"    value="0.1"/>
      <param name="sdf_map/local_map_margin" value="0"/>
      <param name="sdf_map/ground_z"        value="0.0"/>

      <param name="sdf_map/cx" value="3.193751e+02"/>
      <param name="sdf_map/cy" value="2.411233e+02"/>
      <param name="sdf_map/fx" value="3.852785e+02"/>
      <param name="sdf_map/fy" value="3.852785e+02"/>

      <param name="sdf_map/use_depth_filter" value="true"/>
      <param name="sdf_map/depth_filter_tolerance" value="0.15"/>
      <param name="sdf_map/depth_filter_maxdist"   value="0.0"/> <!--4.5-->
      <param name="sdf_map/depth_filter_mindist"   value="-0.2"/> <!--0.2-->
      <param name="sdf_map/depth_filter_margin"    value="2"/>
      <param name="sdf_map/k_depth_scaling_factor" value="1000.0"/>
      <param name="sdf_map/skip_pixel" value="2"/>

      <param name="sdf_map/p_hit"  value="0.65"/>
      <param name="sdf_map/p_miss" value="0.35"/>
      <param name="sdf_map/p_min"  value="0.12"/>
      <param name="sdf_map/p_max"  value="0.90"/>
      <param name="sdf_map/p_occ"  value="0.80"/>
      <param name="sdf_map/min_ray_length" value="0.5"/>
      <param name="sdf_map/max_ray_length" value="5.0"/>

      <param name="sdf_map/slice_height" value="0.3"/>
      <param name="sdf_map/cut_height"   value="2.0"/>
      <param name="sdf_map/show_occ_time"  value="false"/>
      <param name="sdf_map/show_esdf_time" value="false"/>
      <param name="sdf_map/use_uniform_update" value="true"/>

      <param name="sdf_map/pose_type"     value="1"/>  
      <param name="sdf_map/frame_id"      value="map"/>
  </node>
  
  <node pkg = "trr_local_replanner" type = "time_optimal_traj_server" output = "screen" name = "time_optimal_traj_server">
      <remap  from = "~trajectory"        to = "/tr_node/space_time_traj"/>
      <remap  from = "~odometry"          to = "/vins_estimator/imu_propagate"/>
      <remap  from = "~position_command"  to = "/position_cmd"/>
      <remap  from = "~view"              to = "/view" /> 
      <remap  from = "~bspline"           to = "/local_replanner/bspline" />
      <remap  from = "~w_T_odom"          to = "/loop_fusion/pg_T_vio" />
      <remap  from = "~trigger"           to = "/traj_start_trigger" />
      
      <param name="local_planner/traj_order" value="$(arg poly_order)" />
      <param name="local_planner/collision_check_timer"  value="0.1"  /> 
      <param name="local_planner/time_horizon"           value="4"    /> 
      <param name="local_planner/time_commit"            value="0.1"  /> 
      <param name="local_planner/time_lead_yaw"          value="1.0"  /> 
      <param name="local_planner/pub_lead_yaw"           value="true" /> 
      <param name="local_planner/use_lead_yaw_plan1"     value="false"/> 
  </node>

  <!-- <node pkg="waypoint_generator" name="waypoint_generator" type="waypoint_generator" output="screen">
      <remap from="~odom" to="/drone_1/visual_slam/odom"/>        
      <remap from="~goal" to="/goal"/>
      <param name="waypoint_type" value="manual-lonely-waypoint"/>    
  </node>

  <node pkg="trr_global_planner" name="click_obstacle_generation" type="click_obstacle_generation" output="screen">
      <remap from = "~odometry" to = "/drone_1/visual_slam/odom"/>    
      <remap from = "~goal"     to = "/goal"/>

      <param name="map/resolution" value="$(arg resolution)"/>        
      <param name="map/max_height" value="$(arg map_size_z)"/>       

      <param name="ObstacleShape/lower_rad" value="0.2"/>
      <param name="ObstacleShape/upper_rad" value="0.3"/>
      <param name="ObstacleShape/lower_hei" value="2.0"/>
      <param name="ObstacleShape/upper_hei" value="3.0"/>
  </node> -->

</launch>
